controllers/complianceeventsapi/types.go

```diff
@@ -19,6 +22,13 @@ var (
 	errInvalidInput             = errors.New("invalid input")
 )
 
+type dbRow interface {
+	FromRow(*sql.Rows) error
+	GetOrCreate(ctx context.Context, db *sql.DB) error
+	InsertQuery() (string, []any)
+	SelectQuery(returnedColumns ...string) (string, []any)
+}
```

Comment:
I think this interface can/should just be InsertQuery and SelectQuery.

The one current use of the interface is in the `getOrCreate` function, which only uses the methods listed above. It feels especially odd to include GetOrCreate in the interface, because all those implementations use `getOrCreate`, and are in some sense circular.

That said, I'd be fine with accepting the PR as-is.